tests/lib/cpp/cxx/testcase.yaml

```diff
@@ -78,3 +78,11 @@ tests:
     build_only: true
     extra_configs:
       - CONFIG_STD_CPP23=y
+  # Verify struct _cpu_arch struct _thread_arch are compatible with C++ when
+  # specific configs are enabled.
+  cpp.main.empty_struct_size:
+    build_only: true
+    extra_configs:
+      - CONFIG_USERSPACE=n
+      - CONFIG_SMP=n
+      - CONFIG_FPU_SHARING=n
```

Comment:
AFAICS, this is a RISC-V specific issue; so, this should be limited to `arch_allow: riscv`. Also, a RISC-V `integration_platforms` should be added.